PowerShell/JumpCloud Module/Public/Users/Remove-JCUser.ps1

```diff
@@ -67,52 +92,106 @@ UserID has an Alias of _id. This means you can leverage the PowerShell pipeline
-            try {
-                $URI = "$JCUrlBasePath/api/systemusers/$UserID"
-                Write-Warning "Are you sure you wish to delete user: $Username ?" -WarningAction Inquire
-                $delete = Invoke-RestMethod -Method Delete -Uri $URI -Headers $hdrs -UserAgent:(Get-JCUserAgent)
-                $Status = 'Deleted'
-            } catch {
-                $Status = $_.ErrorDetails
-            }
-
-            $FormattedResults = [PSCustomObject]@{
-                'User'    = $Username
-                'Results' = $Status
+            if ($CascadeManager -and $isManager) {
+                Switch ($CascadeManager) {
+                    'NULL' {
+                        $Status = Delete-JCUser -Id $UserID -managerId $null -Headers $hdrs -UserHash $UserHash
+                    }
+                    'Auto' {
+                        if ($hasManagerId) {
+                            $Status = Delete-JCUser -Id $UserID -managerId $hasManagerId -Headers $hdrs -UserHash $UserHash
+                        } else {
+                            $Status = Delete-JCUser -Id $UserID -managerId $null -Headers $hdrs -UserHash $UserHash
+                        }
+                    }
+                    'ID' {
+                        $Status = Delete-JCUser -Id $UserID -managerId $CascadeManagerId -Headers $hdrs -UserHash $UserHash
+                    }
+                }
+            } elseif ($isManager -and !$CascadeManager) {
+                # Prompt for CascadeManager, user enters the ID of the new manager
+                $cascade_manager = Read-Host "User $($Username) is a manager. Do you want to reassign their managed users to another manager? (Y / N)"
+                if ($cascade_manager -eq 'Y') {
+                    if ($hasManagerId) {
+                        $managerUsername = $UserHash.GetEnumerator().Where({ $_.Name -contains ($hasManagerId) }) | Select-Object -ExpandProperty Value | Select-Object -ExpandProperty username
+                        $cascade_manager = Read-Host "User $($Username) is managed by manager: $($managerUsername). Do you want to reassign the managed users to the manager: $($managerUsername)? (Y/N)"
+                        if ($cascade_manager -eq 'Y') {
+                            $newManagerId = $hasManagerId
+                            $Status = Delete-JCUser -Id $UserID -managerId $newManagerId -Headers $hdrs -UserHash $UserHash
+                        } elseif ($cascade_manager -eq 'N') {
+                            $newManagerId = Read-Host "Enter the UserID of the new manager"
+                            # Validate if the Id is a JC User
+                            try {
+                                $validateUser = Get-JcSdkUser -Id $newManagerId
+                                Write-Debug "User $newManagerId is a valid JumpCloud User"
+                                $Status = Delete-JCUser -Id $UserID -managerId $newManagerId -Headers $hdrs -UserHash $UserHash
```

Comment:
I understand why you are running this as a private function `Delete-JCUser` should have it's name changed IMO. 

I'd like our functions to use [PS Approved Verbs](https://learn.microsoft.com/en-us/powershell/scripting/developer/cmdlet/approved-verbs-for-windows-powershell-commands?view=powershell-7.4).

From that link:
"Do not use a synonym of an approved verb. For example, always use `Remove`, never use `Delete` or `Eliminate`."

Which begs the question, why bubble this out into a private function if the name was going to be `Remove-JCUser` anyways. I feel like a better choice might be this:

You have a need to essentially change the URL depending on the actions necessary here. Here are the possible actions and feel free to correct me if I'm wrong:

Two possible options:
1. A user is being deleted and they are not managing other users
2. A user is being deleted and they are managing other users

If they are a manager of other users
1. Set the users subordinates user's manager to be `Null`
2. Set the users subordinates manager to be the `manager` of the user being deleted
3. Set the users subordinates manager to be some new user

This set of code is confusing to me because we have read hosts here, we have read hosts in `Delete-JCUser`. So I'm wondering what the point of the abstraction does here?

If all we need to do is determine if the user being deleted is a manager or not. AND the action we should take IF they are manager. I think this could be simplified into one of two ways:

1. Bring the code back into `Remove-JCUser` and within your tree of conditions, just return a URL string instead of passing the code to `Delete-JCUser`
2. If there's a need to break up the code, create a new function like `Confirm-JCUserForRemoval` that validates that the customers actions will not return some error and return a URL. 

In this way I think the code could be a bit easier to read but let's discuss.

Reply:
I'm of the opinion that this should all be self contained in Remove-JCUser.

Reply:
Read-Host and abstraction wise, we are calling Read-Hosts with our other private functions so I thought it would be okay. The Read-Host calls in Delete-JCUser is only used to confirm the deletion of user, no IDs or usernames being prompted. Remove-JCUser on the other hand, we ask for `Manager ID` or `CascadeManagerId` and validating it in the same function before calling Delete-JCUser to invoke the deletion.

Reply:
Naming convention wise, `Invoke-JCUserRemoval`?

Reply:
Moved Invoke-restmethods and removed the private function